src/lib/auth.js

```diff
@@ -118,10 +118,10 @@ class Auth implements IAuth {
   /**
    * Allow user to access a package.
    */
-  allow_access(packageName: string, user: RemoteUser, callback: Callback) {
+  allow_access({packageName, packageVersion}: AuthPluginPackage, user: RemoteUser, callback: Callback) {
     let plugins = this.plugins.slice(0);
     // $FlowFixMe
-    let pkg = Object.assign({name: packageName}, getMatchedPackagesSpec(packageName, this.config.packages));
+    let pkg = Object.assign({name: packageName, version: packageVersion}, getMatchedPackagesSpec(packageName, this.config.packages));
```

Comment:
In this line I just add new property `version` to `pkg` object. Then [here](https://github.com/verdaccio/verdaccio/pull/1171/files#diff-574051ef1fbe43a2746c5cd241d631c7R133) this object sends to plugins, if I understood correctly. So all previous properties still present, and it should not break current auth plugins.

Reply:
My apologies, you are 100% right. I agree with these changes, good job. But I need you to do something before the merge, we need to base this work on top of `4.x` branch, I'm not accepting features on master branch anymore. In the meantime, I'll provide you with a new version of types to be able to pass the CI.